fortuna/src/config.rs

```diff
@@ -134,6 +134,10 @@ pub struct EthereumConfig {
     /// How many blocks to wait before revealing the random number.
     pub reveal_delay_blocks: BlockNumber,
 
+    /// The provider uri to register on chain.
+    #[serde(default)]
+    pub provider_uri: String,
```

Comment:
two comments about this:
1. it's unfortunate that we will have to repeat the base URL of the service multiple times in the config. We could auto-derive these URIs from the base URL of the service, and that seems ok because this service itself defines the convention for how each chain's URLs is derived from the base.
2. It doesn't make sense for this argument to live in the chain config. Blockchains are not equal to providers. The other provider arguments (like the secret key) are specified elsewhere. 

I think a better approach here would be to make a provider config file that was used specifically by your script to keep on-chain provider info in sync. That config could also hold the fee parameters, for example. In that approach, you wouldn't change register provider or setup provider -- those commands could still be called from the command line as they are right now.